picard/util/pipe.py

```diff
@@ -262,31 +255,32 @@ def send_to_pipe(self, message: str, timeout_secs: Optional[float] = None) -> bo
         :rtype: bool
         """
         if timeout_secs is None:
-            timeout_secs = self.TIMEOUT_SECS
+            timeout_secs = self.TIMEOUT_SECS_WRITE
 
-        with concurrent.futures.ThreadPoolExecutor(max_workers=1) as executor:
-            # we're sending only filepaths, so we have to create some kind of separator
-            # to avoid any potential conflicts and mixing the data
-            sender = executor.submit(self._sender, message + self.MESSAGE_TO_IGNORE)
-            try:
-                if sender.result(timeout=timeout_secs):
-                    return True
-            except concurrent.futures._base.TimeoutError:
-                if self.pipe_running:
-                    log.warning("Couldn't send: %r", message)
-                # hacky way to kill the sender
-                self.read_from_pipe()
-            except Exception as e:
-                # https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.result
-                # If the call raised an exception, this method will raise the same exception.
-                log.error("pipe sender exception: %s", e)
+        # we're sending only filepaths, so we have to create some kind of separator
+        # to avoid any potential conflicts and mixing the data
+        try:
+            sender = self.__thread_pool.submit(
+                self._sender, message + self.MESSAGE_TO_IGNORE)
+            if sender.result(timeout=timeout_secs):
+                return True
+        except concurrent.futures._base.TimeoutError:
+            if self.pipe_running:
+                log.warning("Couldn't send: %r", message)
+            # hacky way to kill the sender
+            self.read_from_pipe()
+        except Exception as e:
+            # https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.result
+            # If the call raised an exception, this method will raise the same exception.
+            log.error("pipe sender exception: %s", e)
 
         return False
 
     def stop(self):
         log.debug("Stopping pipe")
         self.pipe_running = False
         self.send_to_pipe(self.MESSAGE_TO_IGNORE)
+        self.__thread_pool.shutdown(wait=True, cancel_futures=True)
```

Comment:
`cancel_futures` was added in Python 3.9 (see https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown)

Reply:
https://github.com/metabrainz/picard/pull/2252/commits/7a1df3a20dcf203357dc15177b494d61b2ef93f0

Reply:
See https://github.com/metabrainz/picard/pull/2252/commits/7a3ca59ba12048b683e3aa3fb38db6f71d1b41cc

Reply:
See 5b61c2d9077fa3466df9b71b5e02f37b590fc1e8